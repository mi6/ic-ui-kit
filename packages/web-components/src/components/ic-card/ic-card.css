@import "../../global/normalize.css";

:host {
  display: flex;
}

a {
  text-decoration: none !important;
  color: var(--ic-architechtural-black) !important;
}

button {
  border: none;
  background-color: transparent;
  outline: var(--ic-hc-focus-outline);
}

.card,
.card.clickable {
  display: flex;
  flex-direction: column;
  border: var(--ic-border-default);
  border-radius: var(--ic-border-radius);
  box-sizing: border-box;
  padding: var(--ic-space-md);
  text-align: left;
  color: var(--ic-architechtural-black);
  transition: var(--ic-easing-transition-fast);
}

.dark.card,
.dark.card.clickable {
  border: var(--ic-space-1px) solid var(--ic-architectural-700);
}

.card.clickable:hover {
  background-color: var(--ic-action-default-bg-hover);
  border: var(--ic-border-hover);
  cursor: pointer;
}

.card.clickable:focus,
.card.clickable.focussed {
  background-color: var(--ic-action-default-bg-hover);
  box-shadow: var(--ic-border-focus);
  outline: var(--ic-hc-focus-outline);
  border: var(--ic-border-pressed);
}

.card.clickable:active {
  background-color: var(--ic-action-default-bg-active);
  box-shadow: var(--ic-border-focus);
}

.card.disabled {
  border: var(--ic-border-disabled);
}

.card.fullwidth {
  width: 100%;
}

.card-header {
  display: flex;
  align-items: center;
}

.icon {
  display: flex;
  align-items: center;
  padding-right: var(--ic-space-xs);
}

.card.disabled ::slotted(svg) {
  fill: var(--ic-color-tertiary-text);
}

.card.clickable .card-title {
  color: var(--ic-hyperlink);
  text-decoration: underline;
  text-decoration-thickness: var(--ic-space-1px);
}

.card.clickable:hover .card-title,
.card.clickable:focus .card-title,
.card.clickable.focussed .card-title {
  display: inline-block;
  border-bottom: 0.25rem solid !important;
  margin-bottom: -0.25rem !important;
  text-decoration: none;
}

@supports (text-decoration-thickness: 25%) {
  .card.clickable:hover .card-title,
  .card.clickable:focus .card-title,
  .card.clickable.focussed .card-title {
    text-decoration-line: underline;
    text-decoration-thickness: 25%;
    text-underline-offset: 25%;
    border-bottom: 0 !important;
    margin-bottom: 0 !important;
  }
}

.card.clickable:active .card-title {
  text-decoration: none;
}

.card.disabled .card-title {
  text-decoration: underline;
  text-decoration-thickness: var(--ic-space-1px);
  text-decoration-color: var(--ic-color-tertiary-text);
  color: var(--ic-color-tertiary-text);
}

.subheading {
  margin-top: var(--ic-space-xxs);
}

.adornment {
  margin-top: var(--ic-space-xxs);
}

.card-message {
  margin-top: var(--ic-space-md);
  align-items: left;
}

.card.disabled .card-message,
.card.disabled .subheading {
  color: var(--ic-color-tertiary-text);
}

.interaction-button {
  margin-left: auto;
}

.image-top {
  margin-bottom: var(--ic-space-md);
  display: flex;
  justify-content: center;
}

.image-mid {
  margin-top: var(--ic-space-md);
  display: flex;
  justify-content: center;
}

.interaction-area {
  display: flex;
  flex-grow: 1;
  gap: var(--ic-space-md);
  margin-top: var(--ic-space-md);
  align-items: flex-end;
}

.interaction-controls {
  display: flex;
  align-items: center;
  flex-wrap: wrap;
  gap: var(--ic-space-sm);
}

.toggle-button {
  margin-left: auto;
}

.toggle-button-closed svg {
  transform: rotate(90deg);
}

.toggle-button-expanded svg {
  transform: rotate(-90deg);
}

.expanded-content {
  margin-top: var(--ic-space-md);
}
