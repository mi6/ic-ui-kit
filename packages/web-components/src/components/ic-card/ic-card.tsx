import {
  Component,
  Element,
  Listen,
  Prop,
  State,
  h,
  Method,
} from "@stencil/core";
import {
  onComponentRequiredPropUndefined,
  isSlotUsed,
  getThemeFromContext,
  removeDisabledFalse,
} from "../../utils/helpers";
import {
  IcTheme,
  IcThemeForeground,
  IcThemeForegroundEnum,
} from "../../utils/types";
import chevronIcon from "../../assets/chevron-icon.svg";

/**
 * @slot heading - Content will be placed at the top of the card to the right of the icon.
 * @slot message - Content will be placed in the main body of the card.
 * @slot subheading - Content will be placed below the card heading.
 * @slot adornment - Content will be placed below the card subheading.
 * @slot image-top - Content will be placed at the top of the card above all other content.
 * @slot image-mid - Content will be placed below the card heading section.
 * @slot icon - Content will be placed to the left of the card heading.
 * @slot interaction-button - Content will be placed in the top right corner of the heading section.
 * @slot interaction-controls - Content will be placed below the card message.
 * @slot expanded-content - Content will be placed below the interaction controls but will not initially be rendered.
 */
@Component({
  tag: "ic-card",
  styleUrl: "ic-card.css",
  shadow: true,
})
export class Card {
  @Element() el: HTMLIcCardElement;

  @State() appearance?: IcThemeForeground = "default";
  @State() areaExpanded: boolean = false;
  @State() isFocussed: boolean = false;
  @State() parentEl: HTMLElement | null = null;
  @State() parentIsAnchorTag: boolean = false;

  /**
   * If `true`, the card will be a clickable variant, instead of static.
   */
  @Prop({ mutable: true }) clickable?: boolean = false;

  /**
   * If `true`, the card will be disabled if it is clickable.
   */
  @Prop() disabled?: boolean = false;

  /**
   *  If `true`, the card will have an expandable area and expansion toggle button.
   */
  @Prop() expandable?: boolean = false;

  /**
   *  If `true`, the card will fill the width of the container.
   */
  @Prop() fullWidth?: boolean = false;

  /**
   * The heading for the card.
   */
  @Prop() heading!: string;

  /**
   * The URL that the clickable card link points to. If set, the clickable card will render as an "a" tag, otherwise it will render as a button.
   */
  @Prop() href?: string | undefined;

  /**
   * The human language of the linked URL.
   */
  @Prop() hreflang?: string = "";

  /**
   * The main body message of the card.
   */
  @Prop() message?: string = "";

  /**
   * How much of the referrer to send when following the link.
   */
  @Prop() referrerpolicy?: ReferrerPolicy;

  /**
   * The relationship of the linked URL as space-separated link types.
   */
  @Prop() rel?: string;

  /**
   * The subheading for the card.
   */
  @Prop() subheading?: string;

  /**
   * The place to display the linked URL, as the name for a browsing context (a tab, window, or iframe).
   */
  @Prop() target?: string;

  disconnectedCallback(): void {
    if (this.parentIsAnchorTag) {
      this.parentEl.removeEventListener("focus", this.parentFocussed);
      this.parentEl.removeEventListener("blur", this.parentBlurred);
    }
  }

  componentWillLoad(): void {
    this.parentEl = this.el.parentElement;

    if (this.parentEl.tagName === "A") {
      this.clickable = true;
      this.parentIsAnchorTag = true;
      this.parentEl.classList.add("ic-card-wrapper-link");
      this.parentEl.addEventListener("focus", this.parentFocussed);
      this.parentEl.addEventListener("blur", this.parentBlurred);
    }
    removeDisabledFalse(this.disabled, this.el);
  }

  componentDidLoad(): void {
    onComponentRequiredPropUndefined(
      [{ prop: this.heading, propName: "heading" }],
      "Card"
    );
    this.updateTheme();
  }

  @Listen("click", { capture: true })
  handleHostClick(event: Event): void {
    if (this.disabled) {
      event.stopImmediatePropagation();
    }
  }

  @Listen("themeChange", { target: "document" })
  themeChangeHandler(ev: CustomEvent): void {
    const theme: IcTheme = ev.detail;
    this.updateTheme(theme.mode);
  }

  /**
   * Sets focus on the card.
   */
  @Method()
  async setFocus(): Promise<void> {
    if (this.el.shadowRoot.querySelector("a")) {
      this.el.shadowRoot.querySelector("a").focus();
    } else if (this.el.shadowRoot.querySelector("button")) {
      this.el.shadowRoot.querySelector("button").focus();
    }
  }

  private parentFocussed = (): void => {
    this.isFocussed = true;
  };

  private parentBlurred = (): void => {
    this.isFocussed = false;
  };

  private updateTheme(newTheme: IcThemeForeground = null): void {
    const foregroundColor = getThemeFromContext(this.el, newTheme || null);

    if (foregroundColor !== IcThemeForegroundEnum.Default) {
      this.appearance = foregroundColor;
    }
  }

  private toggleExpanded = (): void => {
    this.areaExpanded = !this.areaExpanded;
  };

  render() {
    const {
      clickable,
      disabled,
      expandable,
      heading,
      message,
      href,
      hreflang,
      referrerpolicy,
      rel,
      subheading,
      target,
      fullWidth,
      parentIsAnchorTag,
      isFocussed,
    } = this;

    const Component = parentIsAnchorTag
      ? "div"
      : clickable
      ? this.href === undefined
        ? "button"
        : "a"
      : "div";

    const attrs = Component == "a" && {
      href: href,
      hrefLang: hreflang,
      referrerPolicy: referrerpolicy,
      rel: rel,
      target: target,
    };

    return (
      <Component
        class={{
          ["card"]: true,
          ["clickable"]: clickable && !disabled,
          ["disabled"]: disabled,
          ["fullwidth"]: fullWidth,
          ["focussed"]: isFocussed,
          ["dark"]: this.appearance === IcThemeForegroundEnum.Dark,
        }}
        tabindex={clickable && !parentIsAnchorTag ? 0 : null}
        aria-disabled={disabled ? "true" : null}
        disabled={disabled ? true : null}
        {...attrs}
      >
        {isSlotUsed(this.el, "image-top") && (
          <div class="image-top">
            <slot name="image-top"></slot>
          </div>
        )}
        <div class="card-header">
          {isSlotUsed(this.el, "icon") && (
            <div class="icon">
              <slot name="icon" />
            </div>
          )}
          <div class="card-title">
            <slot name="heading">
              <ic-typography variant="h4">
                <p>{heading}</p>
              </ic-typography>
            </slot>
          </div>
          {isSlotUsed(this.el, "interaction-button") && (
            <div class="interaction-button">
              <slot name="interaction-button"></slot>
            </div>
          )}
        </div>
        {(subheading || isSlotUsed(this.el, "subheading")) && (
          <div class="subheading">
            <slot name="subheading">
              <ic-typography variant="subtitle-small">
                {subheading}
              </ic-typography>
            </slot>
          </div>
        )}
        {isSlotUsed(this.el, "adornment") && (
          <div class="adornment">
            <slot name="adornment"></slot>
          </div>
        )}
        {isSlotUsed(this.el, "image-mid") && (
          <div class="image-mid">
            <slot name="image-mid"></slot>
          </div>
        )}
        {(message || isSlotUsed(this.el, "message")) && (
          <div
            class={{
              ["card-message"]: true,
            }}
          >
            {message && <ic-typography variant="body">{message}</ic-typography>}
            {isSlotUsed(this.el, "message") && <slot name="message"></slot>}
          </div>
        )}
        {(isSlotUsed(this.el, "interaction-controls") || expandable) && (
          <div class="interaction-area">
            <div class="interaction-controls">
              <slot name="interaction-controls"></slot>
            </div>
            {expandable && (
              <ic-button
                class={{
                  ["toggle-button"]: true,
                  ["toggle-button-closed"]: !this.areaExpanded,
                  ["toggle-button-expanded"]: this.areaExpanded,
                }}
                variant="icon"
                size="large"
                disableTooltip
                aria-label="Toggle expandable area"
                aria-expanded={this.areaExpanded}
                aria-controls="expanded-content-area"
                onClick={this.toggleExpanded}
                innerHTML={chevronIcon}
              ></ic-button>
            )}
          </div>
        )}
        {isSlotUsed(this.el, "expanded-content") && this.areaExpanded && (
          <div class="expanded-content" id="expanded-content-area">
            <slot name="expanded-content"></slot>
          </div>
        )}
      </Component>
    );
  }
}
