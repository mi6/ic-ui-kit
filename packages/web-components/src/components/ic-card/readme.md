# ic-card



<!-- Auto Generated Below -->


## Properties

| Property               | Attribute        | Description                                                                                                                                 | Type                                                                                                                                                                                     | Default     |
| ---------------------- | ---------------- | ------------------------------------------------------------------------------------------------------------------------------------------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------- |
| `clickable`            | `clickable`      | If `true`, the card will be a clickable variant, instead of static.                                                                         | `boolean`                                                                                                                                                                                | `false`     |
| `disabled`             | `disabled`       | If `true`, the card will be disabled if it is clickable.                                                                                    | `boolean`                                                                                                                                                                                | `false`     |
| `expandable`           | `expandable`     | If `true`, the card will have an expandable area and expansion toggle button.                                                               | `boolean`                                                                                                                                                                                | `false`     |
| `fullWidth`            | `full-width`     | If `true`, the card will fill the width of the container.                                                                                   | `boolean`                                                                                                                                                                                | `false`     |
| `heading` _(required)_ | `heading`        | The heading for the card.                                                                                                                   | `string`                                                                                                                                                                                 | `undefined` |
| `href`                 | `href`           | The URL that the clickable card link points to. If set, the clickable card will render as an "a" tag, otherwise it will render as a button. | `string`                                                                                                                                                                                 | `undefined` |
| `hreflang`             | `hreflang`       | The human language of the linked URL.                                                                                                       | `string`                                                                                                                                                                                 | `""`        |
| `message`              | `message`        | The main body message of the card.                                                                                                          | `string`                                                                                                                                                                                 | `""`        |
| `referrerpolicy`       | `referrerpolicy` | How much of the referrer to send when following the link.                                                                                   | `"" \| "no-referrer" \| "no-referrer-when-downgrade" \| "origin" \| "origin-when-cross-origin" \| "same-origin" \| "strict-origin" \| "strict-origin-when-cross-origin" \| "unsafe-url"` | `undefined` |
| `rel`                  | `rel`            | The relationship of the linked URL as space-separated link types.                                                                           | `string`                                                                                                                                                                                 | `undefined` |
| `subheading`           | `subheading`     | The subheading for the card.                                                                                                                | `string`                                                                                                                                                                                 | `undefined` |
| `target`               | `target`         | The place to display the linked URL, as the name for a browsing context (a tab, window, or iframe).                                         | `string`                                                                                                                                                                                 | `undefined` |


## Methods

### `setFocus() => Promise<void>`

Sets focus on the card.

#### Returns

Type: `Promise<void>`




## Slots

| Slot                     | Description                                                                               |
| ------------------------ | ----------------------------------------------------------------------------------------- |
| `"adornment"`            | Content will be placed below the card subheading.                                         |
| `"expanded-content"`     | Content will be placed below the interaction controls but will not initially be rendered. |
| `"heading"`              | Content will be placed at the top of the card to the right of the icon.                   |
| `"icon"`                 | Content will be placed to the left of the card heading.                                   |
| `"image-mid"`            | Content will be placed below the card heading section.                                    |
| `"image-top"`            | Content will be placed at the top of the card above all other content.                    |
| `"interaction-button"`   | Content will be placed in the top right corner of the heading section.                    |
| `"interaction-controls"` | Content will be placed below the card message.                                            |
| `"message"`              | Content will be placed in the main body of the card.                                      |
| `"subheading"`           | Content will be placed below the card heading.                                            |


## Dependencies

### Depends on

- [ic-typography](../ic-typography)
- [ic-button](../ic-button)

### Graph
```mermaid
graph TD;
  ic-card --> ic-typography
  ic-card --> ic-button
  ic-button --> ic-loading-indicator
  ic-button --> ic-tooltip
  ic-loading-indicator --> ic-typography
  ic-tooltip --> ic-typography
  style ic-card fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------


