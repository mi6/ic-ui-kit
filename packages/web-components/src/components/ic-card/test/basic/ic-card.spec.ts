import { newSpecPage } from "@stencil/core/testing";
import { Card } from "../../ic-card";

describe("ic-card", () => {
  it("should render", async () => {
    const page = await newSpecPage({
      components: [Card],
      html: `<ic-card heading="Card" message="This is a static card"></ic-card>`,
    });

    expect(page.root).toMatchSnapshot();
  });

  it("should render as a button", async () => {
    const page = await newSpecPage({
      components: [Card],
      html: `<ic-card heading="Card" message="This is a clickable card rendered as a button" clickable=true></ic-card>`,
    });

    expect(page.root).toMatchSnapshot();
  });

  it("should render as a link", async () => {
    const page = await newSpecPage({
      components: [Card],
      html: `<ic-card heading="Card" message="This is a clickable card rendered as a button" clickable=true href="/"></ic-card>`,
    });

    expect(page.root).toMatchSnapshot();
  });

  it("should render full width variant", async () => {
    const page = await newSpecPage({
      components: [Card],
      html: `<ic-card heading="Card" full-width=true message="This is a full width card" clickable=true></ic-card>`,
    });

    expect(page.root).toMatchSnapshot();
  });

  it("should render with a link parent", async () => {
    const page = await newSpecPage({
      components: [Card],
      html: `<a href="/"><ic-card heading="Card" message="This is a clickable card" clickable=true></ic-card></a>`,
    });

    expect(page.root).toMatchSnapshot();
  });

  it("should apply 'focussed' style when parent is focussed", async () => {
    const page = await newSpecPage({
      components: [Card],
      html: `<a href="/"><ic-card id="test-id" heading="Card" message="This is a clickable card" clickable=true></ic-card></a>`,
    });

    expect(page.root).not.toBeNull;

    const element = await document.getElementById("test-id");

    await expect(element.classList.contains("focussed")).toBeFalsy;

    await element.focus();

    await expect(element.classList.contains("focussed")).toBeTruthy;
  });

  it("should lose 'focussed' style when parent loses focus", async () => {
    const page = await newSpecPage({
      components: [Card],
      html: `<a href="/"><ic-card id="test-card" heading="Card" message="This is a clickable card" clickable=true></ic-card></a>`,
    });

    expect(page.root).not.toBeNull;

    const element = await document.getElementById("test-card");

    await element.focus();

    await expect(element.classList.contains("focussed")).toBeTruthy;

    await element.blur();

    await expect(element.classList.contains("focussed")).toBeFalsy;

    await page.rootInstance.disconnectedCallback();
  });

  it("should stop immediate propagation of a click event when disabled", async () => {
    const page = await newSpecPage({
      components: [Card],
      html: `<ic-card id="test-card" heading="Card" message="This is a clickable card" clickable=true disabled=true onclick="alert('test')"></ic-card>`,
    });

    jest.spyOn(window, "alert").mockImplementation();

    const element = await document.getElementById("test-card");

    element.click();

    await page.waitForChanges();

    await expect(window.alert).not.toHaveBeenCalled;
  });

  it("should render with a subheading", async () => {
    const page = await newSpecPage({
      components: [Card],
      html: `<ic-card id="test-card" heading="Card" subheading="Card subheading" message="This is a card"></ic-card>`,
    });

    expect(page.root).toMatchSnapshot();
  });

  it("should render with an interaction button", async () => {
    const page = await newSpecPage({
      components: [Card],
      html: `<ic-card id="test-card" heading="Card" subheading="Card subheading" message="This is a card"><ic-button variant="primary" slot="interaction-button">Click here</ic-button></ic-card>`,
    });

    expect(page.root).toMatchSnapshot();
  });

  it("should render with a top image", async () => {
    const page = await newSpecPage({
      components: [Card],
      html: `<ic-card id="test-card" heading="Card" subheading="Card subheading" message="This is a card"><div slot="image-top">Image placeholder</div></ic-card>`,
    });

    expect(page.root).toMatchSnapshot();
  });

  it("should render with a middle image", async () => {
    const page = await newSpecPage({
      components: [Card],
      html: `<ic-card id="test-card" heading="Card" subheading="Card subheading" message="This is a card"><div slot="image-mid">Image placeholder</div></ic-card>`,
    });

    expect(page.root).toMatchSnapshot();
  });

  it("should render with interaction controls", async () => {
    const page = await newSpecPage({
      components: [Card],
      html: `<ic-card id="test-card" heading="Card" subheading="Card subheading" message="This is a card"><ic-button slot="interaction-controls">Click here</ic-button></ic-card>`,
    });

    expect(page.root).toMatchSnapshot();
  });

  it("should render as expandable", async () => {
    const page = await newSpecPage({
      components: [Card],
      html: `<ic-card id="test-card" heading="Card" subheading="Card subheading" message="This is a card" expandable></ic-card>`,
    });

    expect(page.root).toMatchSnapshot();
  });

  it("should render content in expanded area", async () => {
    const page = await newSpecPage({
      components: [Card],
      html: `<ic-card id="test-card" heading="Card" subheading="Card subheading" message="This is a card" expandable><ic-typography slot="expanded-content">Extra content</ic-typography></ic-card>`,
    });

    page.rootInstance.areaExpanded = true;

    await page.waitForChanges();

    expect(page.root).toMatchSnapshot();
  });

  it("should toggle expanded content when expansion toggle is clicked", async () => {
    const page = await newSpecPage({
      components: [Card],
      html: `<ic-card id="test-card" heading="Card" message="This is a clickable card" expandable><ic-typography slot="expanded-content">Expanded</ic-typography></ic-card>`,
    });

    expect(page.root).toMatchSnapshot();

    page.rootInstance.toggleExpanded();

    await page.waitForChanges();

    expect(page.root).toMatchSnapshot();
  });

  it("should call 'setFocus' when card as a button is focused", async () => {
    const page = await newSpecPage({
      components: [Card],
      html: `<ic-card heading="Card" message="This is a clickable card rendered as a button" clickable=true></ic-card>`,
    });

    //Can't expect anything in this test - this is to increase code coverage only
    await page.rootInstance.setFocus().toHaveBeenCalled;
  });

  it("should call 'setFocus' when card as a link is focused", async () => {
    const page = await newSpecPage({
      components: [Card],
      html: `<ic-card heading="Card" message="This is a clickable card rendered as a button" clickable=true href="/"></ic-card>`,
    });

    //Can't expect anything in this test - this is to increase code coverage only
    await page.rootInstance.setFocus().toHaveBeenCalled;
  });
});
