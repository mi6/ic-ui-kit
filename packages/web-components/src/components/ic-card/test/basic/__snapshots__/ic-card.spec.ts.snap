// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ic-card should render 1`] = `
<ic-card heading="Card" message="This is a static card">
  <mock:shadow-root>
    <div class="card">
      <div class="card-header">
        <div class="card-title">
          <slot name="heading">
            <ic-typography variant="h4">
              <p>
                Card
              </p>
            </ic-typography>
          </slot>
        </div>
      </div>
      <div class="card-message">
        <slot name="message">
          <ic-typography variant="body">
            This is a static card
          </ic-typography>
        </slot>
      </div>
    </div>
  </mock:shadow-root>
</ic-card>
`;

exports[`ic-card should render as a button 1`] = `
<ic-card clickable="true" heading="Card" message="This is a clickable card rendered as a button">
  <mock:shadow-root>
    <button class="card clickable" tabindex="0">
      <div class="card-header">
        <div class="card-title">
          <slot name="heading">
            <ic-typography variant="h4">
              <p>
                Card
              </p>
            </ic-typography>
          </slot>
        </div>
      </div>
      <div class="card-message">
        <slot name="message">
          <ic-typography variant="body">
            This is a clickable card rendered as a button
          </ic-typography>
        </slot>
      </div>
    </button>
  </mock:shadow-root>
</ic-card>
`;

exports[`ic-card should render as a link 1`] = `
<ic-card clickable="true" heading="Card" href="/" message="This is a clickable card rendered as a button">
  <mock:shadow-root>
    <a class="card clickable" href="/" hreflang="" tabindex="0">
      <div class="card-header">
        <div class="card-title">
          <slot name="heading">
            <ic-typography variant="h4">
              <p>
                Card
              </p>
            </ic-typography>
          </slot>
        </div>
      </div>
      <div class="card-message">
        <slot name="message">
          <ic-typography variant="body">
            This is a clickable card rendered as a button
          </ic-typography>
        </slot>
      </div>
    </a>
  </mock:shadow-root>
</ic-card>
`;

exports[`ic-card should render as expandable 1`] = `
<ic-card expandable="" heading="Card" id="test-card" message="This is a card" subheading="Card subheading">
  <mock:shadow-root>
    <div class="card">
      <div class="card-header">
        <div class="card-title">
          <slot name="heading">
            <ic-typography variant="h4">
              <p>
                Card
              </p>
            </ic-typography>
          </slot>
        </div>
      </div>
      <div class="subheading">
        <slot name="subheading">
          <ic-typography variant="subtitle-small">
            Card subheading
          </ic-typography>
        </slot>
      </div>
      <div class="card-message">
        <slot name="message">
          <ic-typography variant="body">
            This is a card
          </ic-typography>
        </slot>
      </div>
      <div class="interaction-area">
        <div class="interaction-controls">
          <slot name="interaction-controls"></slot>
        </div>
        <ic-button aria-controls="expanded-content-area" aria-label="Toggle expandable area" class="toggle-button toggle-button-closed" disabletooltip="" size="large" variant="icon">
          svg
        </ic-button>
      </div>
    </div>
  </mock:shadow-root>
</ic-card>
`;

exports[`ic-card should render content in expanded area 1`] = `
<ic-card expandable="" heading="Card" id="test-card" message="This is a card" subheading="Card subheading">
  <mock:shadow-root>
    <div class="card">
      <div class="card-header">
        <div class="card-title">
          <slot name="heading">
            <ic-typography variant="h4">
              <p>
                Card
              </p>
            </ic-typography>
          </slot>
        </div>
      </div>
      <div class="subheading">
        <slot name="subheading">
          <ic-typography variant="subtitle-small">
            Card subheading
          </ic-typography>
        </slot>
      </div>
      <div class="card-message">
        <slot name="message">
          <ic-typography variant="body">
            This is a card
          </ic-typography>
        </slot>
      </div>
      <div class="interaction-area">
        <div class="interaction-controls">
          <slot name="interaction-controls"></slot>
        </div>
        <ic-button aria-controls="expanded-content-area" aria-expanded="" aria-label="Toggle expandable area" class="toggle-button toggle-button-expanded" disabletooltip="" size="large" variant="icon">
          svg
        </ic-button>
      </div>
      <div class="expanded-content" id="expanded-content-area">
        <slot name="expanded-content"></slot>
      </div>
    </div>
  </mock:shadow-root>
  <ic-typography slot="expanded-content">
    Extra content
  </ic-typography>
</ic-card>
`;

exports[`ic-card should render full width variant 1`] = `
<ic-card clickable="true" full-width="true" heading="Card" message="This is a full width card">
  <mock:shadow-root>
    <button class="card clickable fullwidth" tabindex="0">
      <div class="card-header">
        <div class="card-title">
          <slot name="heading">
            <ic-typography variant="h4">
              <p>
                Card
              </p>
            </ic-typography>
          </slot>
        </div>
      </div>
      <div class="card-message">
        <slot name="message">
          <ic-typography variant="body">
            This is a full width card
          </ic-typography>
        </slot>
      </div>
    </button>
  </mock:shadow-root>
</ic-card>
`;

exports[`ic-card should render with a link parent 1`] = `
<ic-card clickable="true" heading="Card" message="This is a clickable card">
  <mock:shadow-root>
    <div class="card clickable">
      <div class="card-header">
        <div class="card-title">
          <slot name="heading">
            <ic-typography variant="h4">
              <p>
                Card
              </p>
            </ic-typography>
          </slot>
        </div>
      </div>
      <div class="card-message">
        <slot name="message">
          <ic-typography variant="body">
            This is a clickable card
          </ic-typography>
        </slot>
      </div>
    </div>
  </mock:shadow-root>
</ic-card>
`;

exports[`ic-card should render with a middle image 1`] = `
<ic-card heading="Card" id="test-card" message="This is a card" subheading="Card subheading">
  <mock:shadow-root>
    <div class="card">
      <div class="card-header">
        <div class="card-title">
          <slot name="heading">
            <ic-typography variant="h4">
              <p>
                Card
              </p>
            </ic-typography>
          </slot>
        </div>
      </div>
      <div class="subheading">
        <slot name="subheading">
          <ic-typography variant="subtitle-small">
            Card subheading
          </ic-typography>
        </slot>
      </div>
      <div class="image-mid">
        <slot name="image-mid"></slot>
      </div>
      <div class="card-message">
        <slot name="message">
          <ic-typography variant="body">
            This is a card
          </ic-typography>
        </slot>
      </div>
    </div>
  </mock:shadow-root>
  <div slot="image-mid">
    Image placeholder
  </div>
</ic-card>
`;

exports[`ic-card should render with a subheading 1`] = `
<ic-card heading="Card" id="test-card" message="This is a card" subheading="Card subheading">
  <mock:shadow-root>
    <div class="card">
      <div class="card-header">
        <div class="card-title">
          <slot name="heading">
            <ic-typography variant="h4">
              <p>
                Card
              </p>
            </ic-typography>
          </slot>
        </div>
      </div>
      <div class="subheading">
        <slot name="subheading">
          <ic-typography variant="subtitle-small">
            Card subheading
          </ic-typography>
        </slot>
      </div>
      <div class="card-message">
        <slot name="message">
          <ic-typography variant="body">
            This is a card
          </ic-typography>
        </slot>
      </div>
    </div>
  </mock:shadow-root>
</ic-card>
`;

exports[`ic-card should render with a top image 1`] = `
<ic-card heading="Card" id="test-card" message="This is a card" subheading="Card subheading">
  <mock:shadow-root>
    <div class="card">
      <div class="image-top">
        <slot name="image-top"></slot>
      </div>
      <div class="card-header">
        <div class="card-title">
          <slot name="heading">
            <ic-typography variant="h4">
              <p>
                Card
              </p>
            </ic-typography>
          </slot>
        </div>
      </div>
      <div class="subheading">
        <slot name="subheading">
          <ic-typography variant="subtitle-small">
            Card subheading
          </ic-typography>
        </slot>
      </div>
      <div class="card-message">
        <slot name="message">
          <ic-typography variant="body">
            This is a card
          </ic-typography>
        </slot>
      </div>
    </div>
  </mock:shadow-root>
  <div slot="image-top">
    Image placeholder
  </div>
</ic-card>
`;

exports[`ic-card should render with an interaction button 1`] = `
<ic-card heading="Card" id="test-card" message="This is a card" subheading="Card subheading">
  <mock:shadow-root>
    <div class="card">
      <div class="card-header">
        <div class="card-title">
          <slot name="heading">
            <ic-typography variant="h4">
              <p>
                Card
              </p>
            </ic-typography>
          </slot>
        </div>
        <div class="interaction-button">
          <slot name="interaction-button"></slot>
        </div>
      </div>
      <div class="subheading">
        <slot name="subheading">
          <ic-typography variant="subtitle-small">
            Card subheading
          </ic-typography>
        </slot>
      </div>
      <div class="card-message">
        <slot name="message">
          <ic-typography variant="body">
            This is a card
          </ic-typography>
        </slot>
      </div>
    </div>
  </mock:shadow-root>
  <ic-button slot="interaction-button" variant="primary">
    Click here
  </ic-button>
</ic-card>
`;

exports[`ic-card should render with interaction controls 1`] = `
<ic-card heading="Card" id="test-card" message="This is a card" subheading="Card subheading">
  <mock:shadow-root>
    <div class="card">
      <div class="card-header">
        <div class="card-title">
          <slot name="heading">
            <ic-typography variant="h4">
              <p>
                Card
              </p>
            </ic-typography>
          </slot>
        </div>
      </div>
      <div class="subheading">
        <slot name="subheading">
          <ic-typography variant="subtitle-small">
            Card subheading
          </ic-typography>
        </slot>
      </div>
      <div class="card-message">
        <slot name="message">
          <ic-typography variant="body">
            This is a card
          </ic-typography>
        </slot>
      </div>
      <div class="interaction-area">
        <div class="interaction-controls">
          <slot name="interaction-controls"></slot>
        </div>
      </div>
    </div>
  </mock:shadow-root>
  <ic-button slot="interaction-controls">
    Click here
  </ic-button>
</ic-card>
`;

exports[`ic-card should toggle expanded content when expansion toggle is clicked 1`] = `
<ic-card expandable="" heading="Card" id="test-card" message="This is a clickable card">
  <mock:shadow-root>
    <div class="card">
      <div class="card-header">
        <div class="card-title">
          <slot name="heading">
            <ic-typography variant="h4">
              <p>
                Card
              </p>
            </ic-typography>
          </slot>
        </div>
      </div>
      <div class="card-message">
        <slot name="message">
          <ic-typography variant="body">
            This is a clickable card
          </ic-typography>
        </slot>
      </div>
      <div class="interaction-area">
        <div class="interaction-controls">
          <slot name="interaction-controls"></slot>
        </div>
        <ic-button aria-controls="expanded-content-area" aria-label="Toggle expandable area" class="toggle-button toggle-button-closed" disabletooltip="" size="large" variant="icon">
          svg
        </ic-button>
      </div>
    </div>
  </mock:shadow-root>
  <ic-typography slot="expanded-content">
    Expanded
  </ic-typography>
</ic-card>
`;

exports[`ic-card should toggle expanded content when expansion toggle is clicked 2`] = `
<ic-card expandable="" heading="Card" id="test-card" message="This is a clickable card">
  <mock:shadow-root>
    <div class="card">
      <div class="card-header">
        <div class="card-title">
          <slot name="heading">
            <ic-typography variant="h4">
              <p>
                Card
              </p>
            </ic-typography>
          </slot>
        </div>
      </div>
      <div class="card-message">
        <slot name="message">
          <ic-typography variant="body">
            This is a clickable card
          </ic-typography>
        </slot>
      </div>
      <div class="interaction-area">
        <div class="interaction-controls">
          <slot name="interaction-controls"></slot>
        </div>
        <ic-button aria-controls="expanded-content-area" aria-expanded="" aria-label="Toggle expandable area" class="toggle-button toggle-button-expanded" disabletooltip="" size="large" variant="icon">
          svg
        </ic-button>
      </div>
      <div class="expanded-content" id="expanded-content-area">
        <slot name="expanded-content"></slot>
      </div>
    </div>
  </mock:shadow-root>
  <ic-typography slot="expanded-content">
    Expanded
  </ic-typography>
</ic-card>
`;
